src/main/java/io/cdap/plugin/servicenow/sink/service/ServiceNowSinkAPIRequestImpl.java

```diff
@@ -0,0 +1,262 @@
+      apiResponse = restApi.executePost(requestBuilder.build());
+
+      if (!apiResponse.isSuccess()) {
+        LOG.error("Error - {}", getErrorMessage(apiResponse.getResponseBody()));
+        throw new RetryableException();
+      } else {
+        JsonObject responseJSON = jsonParser.parse(apiResponse.getResponseBody()).getAsJsonObject();
+        JsonArray servicedRequestsArray = responseJSON.get(ServiceNowConstants.SERVICED_REQUESTS).getAsJsonArray();
+        JsonElement failedRequestId = null;
+        for (int i = 0; i < servicedRequestsArray.size(); i++) {
+          int statusCode = servicedRequestsArray.get(i).getAsJsonObject().get(ServiceNowConstants.STATUS_CODE)
+            .getAsInt();
+          if (statusCode == HttpStatus.SC_FORBIDDEN || statusCode == HttpStatus.SC_BAD_REQUEST ||
+            statusCode == HttpStatus.SC_INTERNAL_SERVER_ERROR) {
+            String encodedResponseBody = servicedRequestsArray.get(i).getAsJsonObject().get(ServiceNowConstants.BODY)
+              .getAsString();
+            String decodedResponseBody = new String(Base64.getDecoder().decode(encodedResponseBody));
+            String errorDetail = jsonParser.parse(decodedResponseBody).getAsJsonObject().get(ServiceNowConstants.ERROR)
+              .getAsJsonObject().get(ServiceNowConstants.ERROR_DETAIL).getAsString();
+            
+            if (errorDetail.equals(ServiceNowConstants.ACL_EXCEPTION)) {
+              throw new IllegalStateException(String.format("Permission denied for '%s' operation.",
+                                                            config.getOperation()));
+            } else if (errorDetail.contains(ServiceNowConstants.INSERTION_ERROR)) {
+              LOG.warn("Error Response : {} ", decodedResponseBody);
+            } else if (errorDetail.contains((ServiceNowConstants.MAXIMUM_EXECUTION_TIME_EXCEEDED))) {
+              failedRequestId = servicedRequestsArray.get(i).getAsJsonObject().get(ServiceNowConstants.ID);
+            }
+          }
+        }
+
+        JsonArray unservicedRequestsArray = responseJSON.get(ServiceNowConstants.UNSERVICED_REQUESTS).getAsJsonArray();
+
+        if (unservicedRequestsArray.size() > 0) {
+          // Add failed request Id to unserviced requests
+          if (failedRequestId != null) {
+            unservicedRequestsArray.add(failedRequestId);
+          }
+
+          // Process unserviced requests array into unserviced requests map
+          Map<String, RestRequest> unservicedRequestsMap = processUnservicedRequestsArray(restRequestsMap,
+                                                                                          unservicedRequestsArray);
+          // Retry unserviced requests
+          createPostRequest(unservicedRequestsMap, accessToken);
+        }
+      }
+    } catch (IOException e) {
+      LOG.error("Error while connecting to ServiceNow", e.getMessage());
+      throw new RetryableException();
+    } catch (Exception e) {
+      throw e;
```

Comment:
I don't think you need to catch the exception here since you're just throwing it

Reply:
yep, it's unnecessary. Removed it. e8b33614f910fa16f06ffcf185ed0d3a23d9fda1